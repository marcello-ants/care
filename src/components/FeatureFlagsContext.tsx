import { createContext, useContext, ReactNode, useState } from 'react';
import { LDEvaluationDetail } from 'launchdarkly-node-server-sdk';

interface FeatureFlagsProviderProps {
  children: ReactNode;
  flags: FeatureFlags;
}

export type featureFlagsState = {
  flags: FeatureFlags;
};

export type FeatureFlags = Record<string, LDEvaluationDetail | undefined>;

const initialState: featureFlagsState = {
  flags: {},
};

const FeatureFlagsContext = createContext<featureFlagsState>(initialState);

export const FeatureFlagsProvider = ({ children, flags }: FeatureFlagsProviderProps) => {
  const [state] = useState({ flags });
  return <FeatureFlagsContext.Provider value={state}>{children}</FeatureFlagsContext.Provider>;
};

export const useFeatureFlags = () => useContext(FeatureFlagsContext);
