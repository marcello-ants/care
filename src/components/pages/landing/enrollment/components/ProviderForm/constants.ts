export const PROVIDER_SERVICES_DESKTOP = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const PROVIDER_SERVICES_MOBILE = [
  {
    description: 'Child care',
    label: 'Child care',
    value: 'CHILD_CARE',
  },
  {
    description: 'Senior care',
    label: 'Senior care',
    value: 'SENIOR_CARE',
  },
  {
    description: 'Pet care',
    label: 'Pet care',
    value: 'PET_CARE',
  },
  {
    description: 'Tutoring',
    label: 'Tutoring',
    value: 'TUTORING',
  },
  {
    description: 'Special needs care',
    label: 'Special needs care',
    value: 'SPECIAL_NEEDS',
  },
  {
    description: 'Housekeeping',
    label: 'Housekeeping',
    value: 'HOUSEKEEPING',
  },
  {
    description: 'Errands & odd jobs',
    label: 'Errands & odd jobs',
    value: 'GENERIC',
  },
];

export const DROPOFF_URLS = {
  CHILD_CARE: '/swf/ccOptimizedFlow',
  SENIOR_CARE: '/app/enrollment/provider/sc ',
  PET_CARE: '/swf/pcOptimizedFlow',
  HOUSEKEEPING: '/swf/hkOptimizedFlow',
  TUTORING: '/swf/tuOptimizedFlow',
  SPECIAL_NEEDS: '/swf/snOptimizedFlow',
  GENERIC: '/swf/cgOptimizedFlow',
  AU_PAIR: '',
  CARE_GIGS: '',
};
