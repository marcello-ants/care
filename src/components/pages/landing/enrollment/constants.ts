export const SITTER_FORM = {
  providerServices: {
    title: 'What Services do you offer?',
    id: 'providerService',
    options: [
      {
        value: 'CHILD_CARE',
        text: 'Babysitter',
      },
      {
        value: 'CHILD_CARE',
        text: 'Nanny',
      },
      {
        value: 'SPECIAL_NEEDS',
        text: 'Special Needs Provider',
      },
      {
        value: 'TUTORING',
        text: 'Tutor',
      },
      {
        value: 'TUTORING',
        text: 'Private Lesson Instructor',
      },
      {
        value: 'SENIOR_CARE',
        text: 'Senior Care Provider',
      },
      {
        value: 'PET_CARE',
        text: 'Pet Care Provider',
      },
      {
        value: 'HOUSEKEEPING',
        text: 'Housekeeper',
      },
      {
        value: 'GENERIC',
        text: 'Errands &amp; Odd Jobs Provider',
      },
    ],
  },
  providerEmail: {
    id: 'email',
    placeholder: 'Email',
  },
  providerPassword: {
    id: 'password',
    placeholder: 'Password',
  },
  providerZip: {
    id: 'providerZip',
    placeholder: 'ZIP',
  },
};

export const GENERIC_FORM = {
  howDidYouHearAboutUs: {
    title: 'How did you hear about us?',
    id: 'howDidYouHearAboutUs',
    options: [
      {
        value: 'FACEBOOK',
        text: 'Facebook',
      },
      {
        value: 'BANNER_AD',
        text: 'Banner Ad',
      },
      {
        value: 'TV_AD',
        text: 'TV Ad',
      },
      {
        value: 'OTHER_SOCIAL_MEDIA',
        text: 'Other Social Media (Twitter, Instagram, Pinterest, LinkedIn)',
      },
      {
        value: 'PRESS_COVERAGE',
        text: 'Press Coverage (News, Magazine, Blog)',
      },
      {
        value: 'FRIENDS_OR_FAMILY',
        text: 'Friends or Family',
      },
      {
        value: 'ONLINE_VIDEO',
        text: 'Online Video (Hulu, Youtube)',
      },
      {
        value: 'SEARCH_ENGINE',
        text: 'Search Engine (Google, Bing)',
      },
      {
        value: 'PARENTING_GROUP',
        text: 'Parenting Group or Forum',
      },
      {
        value: 'OTHER',
        text: 'Other',
      },
    ],
  },
};
