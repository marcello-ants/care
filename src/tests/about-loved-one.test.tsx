import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { render, cleanup, waitFor, screen } from '@testing-library/react';
import { useRouter } from 'next/router';
import { cloneDeep } from 'lodash-es';
import AboutLovedOne from '@/pages/about-loved-one';
import { FeatureFlags, FeatureFlagsProvider } from '@/components/FeatureFlagsContext';
import { AppStateProvider } from '@/components/AppState';
import { initialAppState } from '@/state';
import { AppState } from '@/types/app';
import GET_TOP_CAREGIVERS from '@/components/request/TopCaregiversGQL';
import { CLIENT_FEATURE_FLAGS } from '@/constants';
import { SeniorCareRecipientRelationshipType, SENIOR_CARE_TYPE } from '@/__generated__/globalTypes';
import userEvent from '@testing-library/user-event';
import AmpliHelper from '@/utilities/ampliAnalyticsHelper';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

const initialStateOverride = cloneDeep(initialAppState);

const topCaregiversMock = {
  request: {
    query: GET_TOP_CAREGIVERS,
    variables: {
      zipcode: '10004',
      serviceID: 'SENIOR_CARE',
      numResults: 6,
      hourlyRate: null,
      hasCareCheck: true,
      qualities: null,
      services: null,
      maxDistanceFromSeeker: null,
    },
  },
  result: {
    data: {
      topCaregivers: [
        {
          caregiver: {
            avgReviewRating: 0,
            numberOfReviews: 0,
            yearsOfExperience: 1,
            hasCareCheck: true,
            responseTime: 8,
            member: {
              address: {
                city: 'Boston',
                state: 'MA',
              },
              displayName: 'Lisa N.',
              firstName: 'Lisa',
              imageURL: 'url1',
              legacyId: '23135',
              id: 'd8aef9f6-a067-4332-a084-606ff368bd4a',
            },
            seniorCareProviderProfile: {
              bio: 'Information',
              services: ['ERRANDS', 'LIGHT_HOUSECLEANING', 'MEAL_PREPARATION'],
              payRange: {
                hourlyRateFrom: {
                  amount: '15',
                  currencyCode: 'USD',
                },
                hourlyRateTo: {
                  amount: '25',
                  currencyCode: 'USD',
                },
              },
              qualities: ['CPR_TRAINED', 'DOES_NOT_SMOKE'],
            },
            signUpDate: null,
            profiles: {
              childCareCaregiverProfile: {
                bio: {
                  experienceSummary: 'bio',
                },
                payRange: {
                  hourlyRateFrom: {
                    amount: '10',
                    currencyCode: 'USD',
                  },
                  hourlyRateTo: {
                    amount: '20',
                    currencyCode: 'USD',
                  },
                },
                qualities: {
                  certifiedNursingAssistant: false,
                  certifiedRegisterNurse: false,
                  certifiedTeacher: false,
                  childDevelopmentAssociate: false,
                  comfortableWithPets: false,
                  cprTrained: false,
                  crn: false,
                  doesNotSmoke: false,
                  doula: false,
                  earlyChildDevelopmentCoursework: false,
                  earlyChildhoodEducation: false,
                  firstAidTraining: false,
                  nafccCertified: false,
                  ownTransportation: false,
                  specialNeedsCare: false,
                  trustlineCertifiedCalifornia: false,
                },
                supportedServices: {
                  carpooling: false,
                  craftAssistance: false,
                  errands: false,
                  groceryShopping: false,
                  laundryAssistance: false,
                  lightHousekeeping: false,
                  mealPreparation: false,
                  swimmingSupervision: false,
                  travel: false,
                },
              },
            },
          },
          distanceFromRequestZip: {
            unit: 'MILES',
            value: 1.2,
          },
        },
      ],
    },
  },
};

const personalizedResult = {
  data: {
    topCaregivers: [
      {
        caregiver: {
          avgReviewRating: 0,
          numberOfReviews: 0,
          yearsOfExperience: 1,
          hasCareCheck: true,
          responseTime: 8,
          member: {
            address: {
              city: 'Boston',
              state: 'MA',
            },
            displayName: 'Lisa N.',
            firstName: 'Lisa',
            imageURL: 'url1',
            legacyId: '23135',
            id: 'd8aef9f6-a067-4332-a084-606ff368bd4a',
          },
          seniorCareProviderProfile: {
            bio: 'Information',
            services: ['ERRANDS', 'LIGHT_HOUSECLEANING', 'MEAL_PREPARATION'],
            payRange: {
              hourlyRateFrom: {
                amount: '15',
                currencyCode: 'USD',
              },
              hourlyRateTo: {
                amount: '25',
                currencyCode: 'USD',
              },
            },
            qualities: ['CPR_TRAINED', 'DOES_NOT_SMOKE'],
          },
          profileURL: 'https://www.care.com/p/lisao9238/sc',
          signUpDate: null,
          profiles: {
            commonCaregiverProfile: {
              id: '1',
            },
            childCareCaregiverProfile: {
              bio: {
                experienceSummary: 'bio',
              },
              qualities: {
                certifiedNursingAssistant: false,
                certifiedRegisterNurse: false,
                certifiedTeacher: false,
                childDevelopmentAssociate: false,
                comfortableWithPets: false,
                cprTrained: false,
                crn: false,
                doesNotSmoke: false,
                doula: false,
                earlyChildDevelopmentCoursework: false,
                earlyChildhoodEducation: false,
                firstAidTraining: false,
                nafccCertified: false,
                ownTransportation: false,
                specialNeedsCare: false,
                trustlineCertifiedCalifornia: false,
              },
              supportedServices: {
                carpooling: false,
                craftAssistance: false,
                errands: false,
                groceryShopping: false,
                laundryAssistance: false,
                lightHousekeeping: false,
                mealPreparation: false,
                swimmingSupervision: false,
                travel: false,
              },
              payRange: {
                hourlyRateFrom: {
                  amount: '10',
                  currencyCode: 'USD',
                },
                hourlyRateTo: {
                  amount: '20',
                  currencyCode: 'USD',
                },
              },
            },
          },
        },
        distanceFromRequestZip: {
          unit: 'MILES',
          value: 1.2,
        },
      },
    ],
  },
};
let personalizedMock10MilesCalled = false;
const topCaregiversPersonalizedMock10Miles = {
  request: {
    query: GET_TOP_CAREGIVERS,
    variables: {
      zipcode: '10004',
      serviceID: 'SENIOR_CARE',
      numResults: 10,
      daysSinceLastActive: 30,
      hasApprovedActivePhoto: true,
      hourlyRate: {
        maximum: { amount: '22', currencyCode: 'USD' },
        minimum: { amount: '14', currencyCode: 'USD' },
      },
      hasCareCheck: true,
      qualities: null,
      services: null,
      isActiveAccount: true,
      maxDistanceFromSeeker: {
        unit: 'MILES',
        value: 10,
      },
    },
  },
  result: () => {
    personalizedMock10MilesCalled = true;
    return personalizedResult;
  },
};

let personalizedMock20MilesCalled = false;
const topCaregiversPersonalizedMock20Miles = {
  request: {
    query: GET_TOP_CAREGIVERS,
    variables: {
      zipcode: '10004',
      serviceID: 'SENIOR_CARE',
      numResults: 10,
      daysSinceLastActive: 30,
      enableFuzzySearch: true,
      hasApprovedActivePhoto: true,
      hourlyRate: {
        maximum: { amount: '22', currencyCode: 'USD' },
        minimum: { amount: '14', currencyCode: 'USD' },
      },
      hasCareCheck: true,
      qualities: null,
      services: null,
      isActiveAccount: true,
      maxDistanceFromSeeker: {
        unit: 'MILES',
        value: 20,
      },
    },
  },
  result: () => {
    personalizedMock20MilesCalled = true;
    return personalizedResult;
  },
};

describe('AboutLovedOne page', () => {
  const mockedWhoNeedsCare = (
    mockWhoNeedsCare = 'SELF',
    mockSelfNeedsCare = false,
    mockComesFromFlow = true,
    // TODO: This is for AB testing and will likely get removed once a flow has been decided
    mockSeekerWhoNeedsCare = 'SELF'
  ) => {
    const override: AppState = {
      ...initialStateOverride,

      seeker: {
        ...initialStateOverride.seeker,
        whoNeedsCare: mockSeekerWhoNeedsCare as SeniorCareRecipientRelationshipType,
        jobPost: {
          ...initialStateOverride.seeker.jobPost,
          lovedOne: {
            ...initialStateOverride.seeker.jobPost.lovedOne,
            whoNeedsCare: mockWhoNeedsCare as SeniorCareRecipientRelationshipType,
          },
          selfNeedsCare: mockSelfNeedsCare,
          comesFromFlow: mockComesFromFlow,
          zip: '10004',
        },
      },
    };

    return override;
  };

  const renderComponent = (initialState: AppState, initialFlagState: FeatureFlags = {}) => {
    return render(
      <FeatureFlagsProvider flags={initialFlagState}>
        <MockedProvider
          mocks={[
            topCaregiversMock,
            topCaregiversPersonalizedMock10Miles,
            topCaregiversPersonalizedMock20Miles,
          ]}>
          <AppStateProvider initialStateOverride={initialState}>
            <AboutLovedOne />
          </AppStateProvider>
        </MockedProvider>
      </FeatureFlagsProvider>
    );
  };

  beforeEach(() => {
    const router = {
      push: jest.fn(),
      pathname: '/about-loved-one/',
    };
    (useRouter as jest.Mock).mockReturnValue(router);

    personalizedMock10MilesCalled = false;
    personalizedMock20MilesCalled = false;
  });

  afterEach(() => {
    cleanup();
  });

  it('should match snapshot', () => {
    const initialState = mockedWhoNeedsCare('SELF', true);
    const { asFragment } = renderComponent(initialState);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display correct copy when both "SELF" and "selfNeedsCare" are both true', () => {
    const initialState = mockedWhoNeedsCare('SELF', true);
    renderComponent(initialState);
    expect(screen.getByText(/Do you/i)).toBeInTheDocument();
    expect(screen.getByText(/about you/i)).toBeInTheDocument();
  });

  it('should display correct copy when "SELF" and "selfNeedsCare" are either true', () => {
    const initialState = mockedWhoNeedsCare('SELF', false);
    renderComponent(initialState);
    expect(screen.getByText(/Do you/i)).toBeInTheDocument();
    expect(screen.getByText(/about you/i)).toBeInTheDocument();
  });

  it('should not display checkboxes when coming from CZEN and selfNeedsCare equals true', () => {
    const initialState = mockedWhoNeedsCare('SELF', true, false);
    renderComponent(initialState);
    expect(screen.queryByText(/Myself/i)).not.toBeInTheDocument();
  });

  it('should display checkboxes when not coming from CZEN', () => {
    const initialState = mockedWhoNeedsCare('SELF', false, true);
    renderComponent(initialState);
    expect(screen.getByText(/Myself/i)).toBeInTheDocument();
  });

  it('should display "Myself" checkbox when coming from CZEN ands selfNeedsCare equals false', () => {
    const initialState = mockedWhoNeedsCare('SELF', false, false);
    renderComponent(initialState);
    expect(screen.queryByText(/Myself/i)).not.toBeInTheDocument();
  });

  it('should call 10 miles then 20 miles for personalized results', async () => {
    expect(personalizedMock10MilesCalled).toEqual(false);
    expect(personalizedMock20MilesCalled).toEqual(false);

    const initialState = mockedWhoNeedsCare('SELF', true);
    const secondaryVariantLCFlagState: FeatureFlags = {
      [CLIENT_FEATURE_FLAGS.LEAD_CONNECT_PROVIDER_NETWORK]: {
        value: 2,
        variationIndex: 2,
        reason: {
          kind: 'secondary test variant',
        },
      },
    };
    renderComponent(initialState, secondaryVariantLCFlagState);
    await new Promise((resolve) => setTimeout(resolve, 0));
    await waitFor(() => expect(personalizedMock10MilesCalled).toEqual(true));
    await waitFor(() => expect(personalizedMock20MilesCalled).toEqual(true));
  });

  it('should trigger both analytic events when feature flag is on', () => {
    const ampliListener = jest.spyOn(AmpliHelper.ampli, 'jobPostedWhoNeedsCare');
    const amplitudeListener = jest.spyOn(AnalyticsHelper, 'logEvent');

    const initialState = mockedWhoNeedsCare('SELF', false, false);
    const inHomeInitialState: Parameters<typeof renderComponent>[0] = {
      ...initialState,
      seeker: {
        ...initialState.seeker,
        typeOfCare: SENIOR_CARE_TYPE.IN_HOME,
      },
    };

    const flags: Parameters<typeof renderComponent>[1] = {
      [CLIENT_FEATURE_FLAGS.AMPLITUDE_USE_AMPLI]: {
        variationIndex: 1,
        value: 'variant',
        reason: { kind: 'FALLTHROUGH' },
      },
    };

    renderComponent(inHomeInitialState, flags);

    const nextButton = screen.getByRole('button', { name: 'Next' });
    userEvent.click(nextButton);

    expect(ampliListener).toBeCalledTimes(1);
    expect(amplitudeListener).toBeCalledTimes(1);
  });
});
