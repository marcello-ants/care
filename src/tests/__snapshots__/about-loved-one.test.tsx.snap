// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutLovedOne page should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root makeStyles-header-16 MuiTypography-h2"
      >
        Share a few details about who needs care.
      </h2>
    </div>
    <span>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h4
          class="MuiTypography-root makeStyles-subheader-12 MuiTypography-h4"
        >
          Who needs care?
        </h4>
        <fieldset
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <div
            aria-label="Who needs care"
            class="MuiFormGroup-root makeStyles-radioGroup-15"
            role="radiogroup"
          >
            <label
              class="MuiFormControlLabel-root makeStyles-radioList-1 MuiFormControlLabel-labelPlacementStart"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    name="who-needs-care"
                    type="radio"
                    value="PARENT"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-35"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-36"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label makeStyles-lovedOneLabel-2 MuiTypography-body1"
              >
                My parent
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root makeStyles-radioList-1 MuiFormControlLabel-labelPlacementStart"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    name="who-needs-care"
                    type="radio"
                    value="SPOUSE"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-35"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-36"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label makeStyles-lovedOneLabel-2 MuiTypography-body1"
              >
                My spouse
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root makeStyles-radioList-1 MuiFormControlLabel-labelPlacementStart"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    name="who-needs-care"
                    type="radio"
                    value="GRANDPARENT"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-35"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-36"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label makeStyles-lovedOneLabel-2 MuiTypography-body1"
              >
                My grandparent
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root makeStyles-radioList-1 MuiFormControlLabel-labelPlacementStart"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-34"
                    name="who-needs-care"
                    type="radio"
                    value="OTHER"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-35"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-36"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label makeStyles-lovedOneLabel-2 MuiTypography-body1"
              >
                My friend/extended relative
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root makeStyles-radioList-1 MuiFormControlLabel-labelPlacementStart"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-31 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-32 Mui-checked MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input-34"
                    name="who-needs-care"
                    type="radio"
                    value="SELF"
                  />
                  <div
                    class="PrivateRadioButtonIcon-root-35 PrivateRadioButtonIcon-checked-37"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-36"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label makeStyles-lovedOneLabel-2 MuiTypography-body1"
              >
                Myself
              </span>
            </label>
          </div>
        </fieldset>
      </div>
    </span>
    <div
      class="MuiGrid-root makeStyles-genderSelector-11 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        class="MuiTypography-root makeStyles-subtitle-13 MuiTypography-h4"
      >
        Gender
      </h4>
      <div
        class="makeStyles-segment-control-38"
      >
        <div
          class="makeStyles-segment-indicator-41"
          id="segment-indicator"
          style="left: 0%; width: 50%;"
        />
        <span
          class="makeStyles-segment-control-option-40 makeStyles-is-selected-42"
          role="button"
          tabindex="0"
        >
          Female
        </span>
        <span
          class="makeStyles-segment-control-option-40"
          role="button"
          tabindex="0"
        >
          Male
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        class="MuiTypography-root makeStyles-subtitle-13 MuiTypography-h4"
      >
        Age
      </h4>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-ageSelector-9"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input makeStyles-inputBox-10"
          role="button"
          tabindex="0"
        >
          80's
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="EIGHTIES"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <h4
      class="MuiTypography-root makeStyles-subheader-12 MuiTypography-h4"
    >
      <strong>
        What should we know about you?
      </strong>
    </h4>
    <div
      class="MuiGrid-root makeStyles-lovedOneContainer-3 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <p
        class="MuiTypography-root makeStyles-examplesText-14 MuiTypography-body2"
      >
        Examples:
      </p>
      <ul
        class="MuiList-root makeStyles-bulletPointList-4 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-bulletPointItem-5"
        >
          <p
            class="MuiTypography-root makeStyles-body3-20 MuiTypography-body1"
          >
            Do you have any underlying conditions?
          </p>
        </li>
        <li
          class="MuiListItem-root makeStyles-bulletPointItem-5"
        >
          <p
            class="MuiTypography-root makeStyles-body3-20 MuiTypography-body1"
          >
            What are your hobbies/interests?
          </p>
        </li>
        <li
          class="MuiListItem-root makeStyles-bulletPointItem-5"
        >
          <p
            class="MuiTypography-root makeStyles-body3-20 MuiTypography-body1"
          >
            How would you like to structure your ideal day?
          </p>
        </li>
      </ul>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <textarea
        class="makeStyles-detailsBox-7"
        id="details"
        name="details"
        placeholder="Share details here"
        rows="7"
        style="height: -4px; overflow: hidden;"
      />
      <textarea
        aria-hidden="true"
        class="makeStyles-detailsBox-7"
        readonly=""
        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
        tabindex="-1"
      />
    </div>
    <div
      class="MuiGrid-root makeStyles-nextButtonContainer-8 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Next
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
